jadx-core/src/main/java/jadx/core/dex/visitors/SimplifyVisitor.java

```diff
@@ -95,6 +98,44 @@ private static InsnNode simplifyInsn(MethodNode mth, InsnNode insn) {
 				}
 				break;
 
+			case CONSTRUCTOR:
+				ConstructorInsn constructor = (ConstructorInsn) insn;
+				if (constructor.getCallMth().getDeclClass().getType().equals(ArgType.STRING)) {
+					InsnArg arg = insn.getArg(0);
```

Comment:
Check args count before getting first. Now fail on `new String()`
Also check type of arg, because there are many string constructors.
Or better check constructor signature (`constructor.getCallMth().getShortId()`)
Also you can easily support constructor with char array: `new String(new char[]{'a', 'b', 'c'});`

Reply:
Type type is checked later (byte or newly added char)